packages/node/src/retry/defaultRetry.ts

```diff
@@ -234,11 +235,13 @@ export class RetryHandler extends BaseRetryHandler {
       await asyncSleep(sleepDuration);
       const isLastTry = numRetries === this._options.retryTimeouts.length;
       const eventsToRetry = eventsBuffer.slice(0, eventCount);
-      const { shouldRetry, shouldReduceEventCount, eventIndicesToRemove } = await this._retryEventsOnce(
+      const { shouldRetry, shouldReduceEventCount, eventIndicesToRemove, response } = await this._retryEventsOnce(
         userId,
         deviceId,
         eventsToRetry,
       );
+      if (this._options.onRetry !== null && response.status !== Status.Success)
+        this._options.onRetry(response, numRetries);
```

Comment:
Also pass in `isLastTry` as an arg, so users have a quick way to know if the callback will be called again